### DISCUSSION TITLE
Recovery Image in PDF bug

### DISCUSSION THREAD

**Discussion by ericosmic** [pymupdf/PyMuPDF#2492] [Looking for help] [answered]
os system : ubuntu 
fitz version: ('1.20.2', '1.20.3', '20220813000001')

bug describtion: PDF file contain some images ,  but extracted info of this image only contain color not words.  The pdf attach below:
[zf1.pdf](https://github.com/pymupdf/PyMuPDF/files/11858955/zf1.pdf)

**Comment by JorjMcKie** [comment, marked as answer]
Page 0 contains 4 images with masks, where each image and its mask have different resolutions.
PyMuPDF does not support this.
```python
In [1]: import fitz
In [2]: doc=fitz.open("zf1.pdf")
In [3]: page=doc[0]
In [4]: page.get_images()
Out[4]:
[(28, 29, 2, 2, 1, 'Indexed', '', 'Image28', ''),
 (39, 40, 2, 2, 1, 'Indexed', '', 'Image39', ''),
 (41, 42, 2, 2, 1, 'Indexed', '', 'Image41', ''),
 (43, 44, 2, 2, 1, 'Indexed', '', 'Image43', '')]
In [5]: print(doc.xref_object(28)) # base image
<<
  /Type /XObject
  /Subtype /Image
  /Width 2  # <===
  /Height 2 # <===
  /ColorSpace [ /Indexed /DeviceRGB 1 <FF0000FFFFFF> ]
  /BitsPerComponent 1
  /Interpolate false
  /SMask 29 0 R
  /Length 2
>>
In [6]: print(doc.xref_object(29))  # SMask
<<
  /Type /XObject
  /Subtype /Image
  /Width 3186 # <===
  /Height 401 # <===
  /ColorSpace /DeviceGray
  /BitsPerComponent 1
  /Filter /FlateDecode
  /Length 10321
>>
```
There is no way in PyMuPDF to combine base image and its SMask in this case.
You can however extract for instance xref 29 as a separate image and save it as a PNG, which will then also show the text visible on page.

**Reply by ericosmic** [reply]
Thanks for your reply. It is big help.
